addons/website_forum/models/forum_post.py

```diff
@@ -515,13 +514,20 @@ def close(self, reason_id):
                     _('Post is closed and marked as offensive content')
                 )
                 post.create_uid.sudo()._add_karma(karma, post, message)
-
-        self.write({
-            'state': 'close',
-            'closed_uid': self._uid,
-            'closed_date': datetime.today().strftime(tools.DEFAULT_SERVER_DATETIME_FORMAT),
-            'closed_reason_id': reason_id,
-        })
+        if reason.reason_type == 'offensive':
+            self.write({
+                'state': 'offensive',
+                'closed_uid': self._uid,
+                'closed_date': datetime.today().strftime(tools.DEFAULT_SERVER_DATETIME_FORMAT),
```

Comment:
why do you convert it to string ?

Reply:
I have not changed this part, 'closed_date' was already being converted to string before.